File: Praticas/05/Table/Table.java
public class Table {
    private int _ints[];
    
    public Table(int n){
        _ints = new int[n];
    }

    public int getValue(int pos){
        return _ints[pos];
    }

    public void setValue(int pos, int value){
        _ints[pos]=value;
    }

    public void setAll(int value){
        for(int i=0; i<_ints.length; i++){
            _ints[i]=value;
        }
    }

    public boolean contains(SelectionPredicate p){
        for(int i=0; i<_ints.length; i++){
            if(p.ok(_ints[i])){
                return true;
            }
        }
        return false;
    }
}

File: Praticas/c++/05/Table.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Printer{
public:
    virtual string format(int i) = 0;
};


class Table{
    vector<int> _vector;
public:
    Table(int s){
        _vector = vector<int>(s, 0);
    }

    int getValue(int pos){
        return _vector.at(pos);
    }

    void setValue(int pos, int value){
        _vector.at(pos) = value;
    }

    void setAll(int value){
        for(int i=0; i<_vector.size(); i++){
            _vector.at(i) = value;
        }
    }

    void print(Printer& p){
        for(int i=0; i<_vector.size(); i++){
            cout << p.format(_vector.at(i)) << endl;
        }
    }
};


class BracketsPrinter: public Printer{
public:
    virtual string format(int i) override {
        return "{" + to_string(i) + "}";
    }
};

class DuplicatePrinter: public Printer{
public:
    virtual string format(int i) override {
        return to_string(i) +" "+ to_string(i);
    }
};

int main(){
    Table* t = new Table(10);

    t->setAll(5);

    BracketsPrinter* b = new BracketsPrinter();
    DuplicatePrinter* d = new DuplicatePrinter();

    t->print(*b);
    t->print(*d);

    delete t;
    delete d;
    delete b;

    return 0;
}